salt/states/pkg.py

```diff
@@ -1725,6 +1890,247 @@ def installed(
+        return targets
+    elif not isinstance(targets, dict):
+        return {'name': name,
+                'changes': {},
+                'result': False,
+                'comment': 'An error was encountered while checking targets: '
+                           '{0}'.format(targets)}
+
+    comment = []
+    if __opts__['test']:
+        summary = ', '.join(targets)
+        comment.append('The following packages would be '
+                       'downloaded: {0}'.format(summary))
+        ret = {'name': name,
+               'changes': {},
+               'result': None,
+               'comment': '\n'.join(comment)}
+        return ret
+
+    changes = {}
+    try:
+        pkg_ret = __salt__['pkg.install'](name=name,
+                                          pkgs=pkgs,
+                                          version=version,
+                                          downloadonly=True,
+                                          fromrepo=fromrepo,
+                                          ignore_epoch=ignore_epoch,
+                                          **kwargs)
+        changes.update(pkg_ret)
+    except CommandExecutionError as exc:
+        ret = {'name': name, 'result': False}
+        if exc.info:
+            # Get information for state return from the exception.
+            ret['changes'] = exc.info.get('changes', {})
+            ret['comment'] = exc.strerror_without_changes
+        else:
+            ret['changes'] = {}
+            ret['comment'] = ('An error was encountered while downloading '
+                              'package(s): {0}'.format(exc))
+        return ret
+
+    new_pkgs = __salt__['pkg.list_downloaded']()
+    ok, failed = _verify_install(targets, new_pkgs, ignore_epoch=ignore_epoch)
+
+    result = True
+    if failed:
+        summary = ', '.join([_get_desired_pkg(x, targets)
+                             for x in failed])
+        comment.insert(0, 'The following packages failed to '
+                          'download: {0}'.format(summary))
+        result = False
+
+    if not changes and not comment:
+        comment.append('Packages are already downloaded: '
+                       '{0}'.format(', '.join([_get_desired_pkg(x, targets)])))
+
+    ret = {'name': name,
+           'changes': changes,
+           'result': result,
+           'comment': '\n'.join(comment)}
+    return ret
+
+
+def patch_installed(name, advisory_ids=None, downloadonly=None, **kwargs):
+    '''
+    Ensure that packages related to certain advisory ids are installed.
+
+    Currently supported for the following pkg providers:
+    :mod:`yumpkg <salt.modules.yumpkg>` and :mod:`zypper <salt.modules.zypper>`
+
+    CLI Example:
+
+    .. code-block:: yaml
+
+        issue-foo-fixed:
+          pkg.patch_installed:
+            - advisory_ids:
+              - SUSE-SLE-SERVER-12-SP2-2017-185
+              - SUSE-SLE-SERVER-12-SP2-2017-150
+              - SUSE-SLE-SERVER-12-SP2-2017-120
+    '''
```

Comment:
Same as with ``pkg.downloaded``, let's gate this state by checking for ``pkg.list_patches`` in the ``__salt__`` dunder.